keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/IntegerVertexSamplesTest.java

```diff
@@ -0,0 +1,81 @@
+package io.improbable.keanu.vertices.dbl;
+
+import com.google.common.collect.ImmutableList;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.intgr.IntegerVertexSamples;
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class IntegerVertexSamplesTest {
+    List<IntegerTensor> values = ImmutableList.of(
+        IntegerTensor.create(new int[]{0, 16, 4}, 1, 3),
+        IntegerTensor.create(new int[]{-4, -8, 4}, 1, 3),
+        IntegerTensor.create(new int[]{8, -4, 12}, 1, 3),
+        IntegerTensor.create(new int[]{4, 4, 8}, 1, 3)
+    );
+    final IntegerVertexSamples samples = new IntegerVertexSamples(values);
+
+    List<SummaryStatistics> stats = ImmutableList.of(
+        new SummaryStatistics(),
+        new SummaryStatistics(),
+        new SummaryStatistics()
+    );
+
+    @Before
+    public void calculateSummaryStatistics() {
+        for (IntegerTensor tensor : values) {
+            for (int i = 0; i < tensor.getLength(); i++) {
+                stats.get(i).addValue(tensor.getValue(i));
+            }
+        }
+    }
+
+    @Test
+    public void doesCalculateAverage() {
+        DoubleTensor averages = samples.getAverages();
+
+        double[] expectedValues = stats.stream().mapToDouble(SummaryStatistics::getMean).toArray();
+        assertThat(averages.asFlatDoubleArray(), equalTo(expectedValues));
+        assertThat(averages.asFlatDoubleArray(), equalTo(new double[] {2.0, 2.0, 7.0}));
+    }
+
+    @Test
+    public void doesGetScalarMode() {
+        int mode = samples.getScalarMode();
+        assertThat(mode, equalTo(0));
+    }
+
+    @Test
+    public void getModeAtIndexReturnsMostFrequentElement() {
+        int mode = samples.getModeAtIndex(0, 2);
+        assertThat(mode, equalTo(4));
+    }
+
+    @Test
+    public void getModeAtIndexReturnsFirstElementIfAllUnique() {
+        int mode = samples.getModeAtIndex(0, 1);
+        assertThat(mode, equalTo(16));
+    }
+
+    @Test
+    public void canGetSamplesAsTensor() {
+        List<IntegerTensor> samplesAsList = samples.asList();
+
+        IntegerTensor samplesAsTensor = samples.asTensor();
+        List<IntegerTensor> samplesAsTensorSliced = samplesAsTensor.sliceAlongDimension(0, 0, samplesAsTensor.getShape()[0]);
```

Comment:
Similar to my point above, you can use split to do this.

Reply:
Not implemented for Integer tensors either, so I don't think we gain much? And I think slicing the entire first dimension makes more sense?

Reply:
Good point. I am not keen on adding `sliceAlongDimension` to `NumberTensor` at the moment if it only is being used in this test. See what everyone else thinks but I think it would be better for the `sliceAlongDimension` to sit with the test.

Reply:
That makes sense @wesleyw72 .

`sliceAlongDimension` is currently only used in one place (`Multinomial`). Can we replace that with a `split` too?